virttest/utils_windows/drive.py

```diff
@@ -37,3 +39,93 @@ def get_floppy_drives_letter(session):
     """
     cond = "MediaType!=0 AND MediaType!=11 AND MediaType!=12"
     return _logical_disks(session, cond=cond, props=["DeviceID"])
+
+
+def rescan_disk(session):
+    """
+    Rescan disk in windows guest.
+
+    :param session: Session object.
+    """
+    rescan_cmd = "echo rescan > cmd"
+    rescan_cmd += " && echo exit >> cmd"
+    rescan_cmd += " && diskpart /s cmd"
+    session.cmd(rescan_cmd)
+
+
+def get_drive_volume_id(session, did, disk_letter):
+    """
+    Get a drive volume id in windows guest.
+
+    :param session: Session object.
+    :param did: Disk index.
+    :param disk_letter: Destination volume location.
+
+    :return: volume id, cmd header and cmd footer
+    """
+    list_disk_cmd = "echo list disk > cmd && "
+    list_disk_cmd += "echo exit >> disk && diskpart /s cmd"
+    disks = session.cmd_output(list_disk_cmd, timeout=120)
+
+    for disk in disks.splitlines():
+        if re.search(r"DISK %s" % did, disk, re.I | re.M):
+            cmd_header = 'echo list disk > cmd &&'
+            cmd_header += ' echo select disk %s >> cmd &&' % did
+            cmd_footer = ' echo exit >> cmd && diskpart /s cmd'
+            cmd_footer += ' && del /f cmd'
+            detail_cmd = ' echo detail disk >> cmd &&'
+            detail_cmd = ' '.join([cmd_header, detail_cmd, cmd_footer])
+            logging.debug("Detail for 'Disk%s'" % did)
+            details = session.cmd_output(detail_cmd)
+            volume_id = re.findall("Volume (\d)\s+%s" % disk_letter, details, re.M)[0]
+    return volume_id, cmd_header, cmd_footer
+
+
+def extend_partition_windows(session, did, disk_letter):
+    """
+    Extend a partition in windows guest.
+    Default extend to max usage size.
+
+    :param session: Session object.
+    :param did: Disk index.
+    :param disk_letter: Disk letter to be extended.
+    """
+    (volume_id, cmd_header, cmd_footer) = get_drive_volume_id(session,
+                                                              did, disk_letter)
+    cmd_header += ' echo select volume %s >> cmd && ' % volume_id
+    extend_cmd = ' echo extend >> cmd &&'
```

Comment:
Considering about concurrency, it may have problems if only use the unique script file `cmd`. Please take a look at the following instance to get the right usage.
https://github.com/avocado-framework/avocado-vt/blob/8a69648201400539971897e5fe7eef0bc185677c/virttest/utils_disk.py#L261-L270

Reply:
re-define the script name, and replace with temp file~
thanks~